### ISSUE TITLE
Namespace controller causes unnecessary reconciliation of metrics module

### ISSUE THREAD

**Issue by albatross0** [microsoft/retina#233]
**Describe the bug**
The metrics module repeats reconciliation.

**To Reproduce**

1. kubectl create ns test1
2. kubectl create ns test2
3. kubectl annotate ns test1 retina.sh=observe
4. kubectl annotate ns test2 retina.sh=observe

**Expected behavior**
The metrics module will not reconcile until MetricsSpec changes.

**Platform (please complete the following information):**

- OS: Rocky Linux 9
- Kubernetes Version: 1.28.8
- Host: self-host
- Retina Version: v0.0.2

**Additional context**

```
ts=2024-04-05T15:08:45.262Z level=info caller=metrics/metrics_module.go:156 msg="Reconciling metric module" goversion=go1.21.8 os=linux arch=amd64 numcores=4 hostname=k3s03.localdomain podname=retina-agent-mfgzk version=v0.0.2 apiserver=https://10.43.0.1:443 plugins=dropreason,packetforward,linuxutil,dns,packetparser spec= specError="unsupported value type"
ts=2024-04-05T15:08:45.262Z level=info caller=metrics/metrics_module.go:182 msg="Including namespaces" goversion=go1.21.8 os=linux arch=amd64 numcores=4 hostname=k3s03.localdomain podname=retina-agent-mfgzk version=v0.0.2 apiserver=https://10.43.0.1:443 plugins=dropreason,packetforward,linuxutil,dns,packetparser namespaces=test2,test1

ts=2024-04-05T15:08:46.262Z level=info caller=metrics/metrics_module.go:156 msg="Reconciling metric module" goversion=go1.21.8 os=linux arch=amd64 numcores=4 hostname=k3s03.localdomain podname=retina-agent-mfgzk version=v0.0.2 apiserver=https://10.43.0.1:443 plugins=dropreason,packetforward,linuxutil,dns,packetparser spec= specError="unsupported value type"
ts=2024-04-05T15:08:46.262Z level=info caller=metrics/metrics_module.go:182 msg="Including namespaces" goversion=go1.21.8 os=linux arch=amd64 numcores=4 hostname=k3s03.localdomain podname=retina-agent-mfgzk version=v0.0.2 apiserver=https://10.43.0.1:443 plugins=dropreason,packetforward,linuxutil,dns,packetparser namespaces=test1,test2
```

The elements of the namespace array returned by GetAnnotatedNamespaces() are in random order, so
If there are more than one annotated namespace, the namespace controller may call Reconcile() with a different spec than before.

**Comment by rbtr** [comment]
@albatross0 it looks like you're already prepping a fix, can I assign this to you?

**Comment by albatross0** [comment]
@rbtr Yes. I fixed this issue in my environment.

**Comment by rbtr** [comment]
fixed in #234
